docs/devguide/modules-dev-guide.asciidoc

```diff
@@ -485,8 +485,45 @@ run.
 In order to test the filesets with the sample logs and/or generate the expected output one should run the tests
 locally for a specific module, using the following procedure under Filebeat directory:
 
-. Run an Elasticsearch instance locally using docker: `docker run -p 9200:9200 -p 9300:9300 -e "discovery.type=single-node" docker.elastic.co/elasticsearch/elasticsearch:8.0.0-SNAPSHOT`
+. Start an Elasticsearch instance locally. For example, using Docker:
++
+[source,bash]
+----
+docker run \
+  --name elasticsearch \
+  -p 9200:9200 -p 9300:9300 \
+  -e "xpack.security.http.ssl.enabled=false"  -e "ELASTIC_PASSWORD=changeme" \
+  -e "discovery.type=single-node" \
+  --pull allways --rm --detach \
+  docker.elastic.co/elasticsearch/elasticsearch:master-SNAPSHOT
+----
+. Create an "admin" user on that Elasticsearch instance:
++
+[source,bash]
+----
+curl -u elastic:changeme \
+  http://localhost:9200/_security/user/admin \
+  -X POST -H 'Content-Type: application/json' \
+  -d '{"password": "changeme", "roles": ["superuser"]}'
+----
+. Create the testing binary: `make filebeat.test`
+. Update fields yaml: `make update`
 . Create python env: `make python-env`
 . Source python env: `source ./build/python-env/bin/activate`
-. Create the testing binary: `make filebeat.test`
-. Run the test, ie: `GENERATE=1 INTEGRATION_TESTS=1 BEAT_STRICT_PERMS=false TESTING_FILEBEAT_MODULES=nginx pytest tests/system/test_modules.py`
+. Run a test, for example to check nginx access log parsing:
++
+[source,bash]
+----
+INTEGRATION_TESTS=1 BEAT_STRICT_PERMS=false ES_PASS=changeme \
+TESTING_FILEBEAT_MODULES=nginx \
+pytest tests/system/test_modules.py -v --full-trace
+----
+. Add and remove option env vars as required. Here are some useful ones:
+* `TESTING_FILEBEAT_MODULES`: comma separated list of modules to test.
```

Comment:
Great to see this documented here. @ph @cmacknz This solves the problem I mentioned in another PR.

Reply:
Maybe it is worth documenting `TESTING_FILEBEAT_ALLOW_OLDER` as well with a note that we do not expect all features to be backwards compatible.

Reply:
![Screen Shot 2022-02-01 at 11 43 36](https://user-images.githubusercontent.com/690/151905743-a91bd94c-0495-4f77-8672-89589c0e3a64.png)

now to go see if autopilot got it right 😆

Reply:
Nope!

Reply:
Updated the old fashion (human) way :)